man/ncid-page.1
.\" %W% %G%
.TH NCID-PAGE 1
.SH NAME
.B ncid-page
.SH SYNOPSIS
ncid --no-gui --program
.B ncid-page
.SH DESCRIPTION
The
.B ncid-page
output module sends the Caller ID or message to a cell phone, pager,
or any other email address.  Can have multiple email addresses
.PP
It normally sends the information as soon as the Caller ID
is received.  It can be configured to wait until a specific number of rings
before sending the Caller ID when using a modem that indicates RING, or it
can send the Caller ID if the call was unanswered or at the completion of
the call when using a SIP gateway.  A message is always sent immediately.
.PP
The module requires \fIPageTo\fR in \fIncid-page.conf\fR to be set
to the SMS gateway for your cell phone carrier.
.TP 10
List of SMS gateways:
https://en.wikipedia.org/wiki/List_of_SMS_gateways
.SH REQUIREMENTS
mail program
.SH EXAMPLES
If you want to call a module ona certain ring count, you need
to configure the module to set ring in the ncid.conf file.
.PP
The \fIncid-page\fR module requires the \fIPageTo\fR variable to be set.
The \fIncid-notify\fR module requires either a key variable to be set
to a API key or a key file that contains the API key
key file
.PP
.RS 0
ncid command line options to call ncid-page at ring number 4

    ncid --no-gui --program ncid-page --ring 4

ncid.conf line and ncid command line to call ncid-page if no answer

    uncomment the ncid_page line in ncid.conf and change 4 to -1
    set ncid_page {set Ring -1}

    ncid command to call ncid-page if no answer
    ncid --no-gui --program  ncid-page
.RE
.PP
.SH FILES
/etc/ncid/conf.d/ncid-page.conf
.br
/usr/share/ncid/ncid-page
.SH SEE ALSO
ncidd.8,
ncid-modules.7,
ncid.1,
ncid-alert.1,
ncid-initmodem.1,
ncid-kpopup.1,
ncid-mythtv.1,
ncid-notify.1,
ncid-samba.1,
ncid-skel.1,
ncid-speak.1,
ncid-wakeup.1,
ncid-yac.1


doc/CHANGES
ncid - Network CallerID

Version 0.87
    - README: updated
    - Makefile: added test: $(DOCDIR) does not exist, create it
    - All GPL license headers in files were updated to GPL 3
    - client/ncid.sh: logClock{} changed {^(CID|HUP|OUT)} to {^(?:CID|HUP|OUT)}
                      added "Line Labels" in help menu and labInfo {}
                      checkTypes() added new types and changed type numbers
                      getCID(): added new types and changed if statements
                                for new Type numbers
    - debian/Makefile: updated
    - debian/README.Debian: updated
    - debian/*.init: removed "S" from default STOP
    - debian/wc2ncid.init: removed configuration stuff, changed sbin to sbin
    - debian/yac2ncid.init: removed configuration stuff, changed sbin to sbin
    - debian/rn2ncid.init: new
    - doc/INSTALL-Ubuntu: updated
    - doc/INSTALL-Raspbian: updated
    - doc/LICENSE: updated to GPL 3
    - doc/Setup-rn2ncid: new
    - doc/Verbose-rn2ncid: new
    - Fedora/Makefile: updated
    - Fedora/README.Fedora: updated
    - Fedora/rn2ncid.init: new
    - Fedora/rn2ncid.service: new
    - FreeBSD/Makefile: updated
    - FreeBSD/README.FreeBSD: updated
    - FreeBSD/rn2ncid.rc: new
    - gateway/Makefile: added $(EXTRA_CFLAGS) to CFLAGS definition
    - gateway/README.Gateways: renamed to README.gateways and updated
    - gateway/*.[ch]: updated license in headers
    - gateway/*.pl: updated license in comments
    - gateway/sip2ncid.h: added UNAUTHORIZED define
    - gateway/sip2ncid.c: processPackets() added UNAUTHORIZED section
    - gateway/wc2ncid.conf: - improved comment wording
                            - changed nciddelay to delay
    - gateway/wc2ncid.pl: - moved test mode just after Command Line Processing
                          - moved logfile and config file messages to
                            after start message
                          - added cleanup(), called by errorExit() & sigHandle()
                          - added delay option to pod documentation wording,
                            improved some pod wording
                          - changed $nciddelay to $delay, added $Delay,
                            added --delay|-d option
    - gateway/rn2ncid.pl: new
    - gateway/rn2ncid.config: new
    - Mac/Makefile: updated
    - Mac/README.Mac: updated
    - Mac/net.sourceforge.rn2ncid.plist: new
    - man/Makefile: updated
    - man/ncid-*.1: updated
    - man/ncid.1: updated
    - man/ncid.conf.5: updated
    - man/ncidd.8: updated
    - man/yac2ncid.1: updated
    - man/rn2ncid.conf.5: new
    - modules/ncid-*.sh: updated
    - modules/ncid-*.conf: updated
    - modules/ncid-notify.sh: changed line one from /bin/sh to /bin/bash
                              minor changes to verbose output and comments
    - modules/ncid-speak.sh: added "AreaCodeLength" code to speak the area
                             code instead or "NO NAME" or entire number
    - modules/ncid-speak.conf-in: added "AreaCodeLength" and comments
    - server/ncidd.c: writeLog() - moved "logMsg(LEVEL3, msgbuf);" to start
                      formatCID() - moved cidnoname test inside
                                    if (!(cid.status & CIDNMBR))
                                 - added PID:
                      added BLKLINE, NOTLINE, PIDLINE to *lineTags[]
                      doPoll: - added code for a NOT: message
    - server/ncidd.h: added defines for PID, PIDLINE, NOTLINE, CALLPID
    - test/README.test: updated
    - test/*: renamed test programs and test data
    - test/test-rn2ncid: new
    - test/test-rn2ncid.data: new
    - TiVo/ncid-tivo.sh:  if no line label, display $CIDNAME/$CIDNMBR
                          else display "$CIDNAME $CIDNMBR/$CIDTYPE $CIDLINE"

Version 0.86.1
    - debian/changelog: updated
    - debian/control: for package ncid-client added "curl" to Depends

Version 0.86
    - README: updated
              added redhat: and redhat-install:
    - attic/*: new
    - client/ncid.sh: added more doVerbose statements and levels
                      changed 300* doVerbose message to any one of four
                      added line count to "End of call log" message
                      removed the --raw | -R option
                      fixed and improved call on a ring number
                      fonts used depend on platform and OS
                      allow resizing of window
                      allow font size changes
                      time format in history window can be either 12 and 24
                      new menu called Preferences for Fonts and time format
                      menu item to save window size
                      menu item to save window size and location
                      added the "-h | --help" option
                      added the "-v | --version" option
                      added the "-w | --wakeup" option
                      getCID{}: added calls to ncid-wakeup module
                      changed "-V" to be the "-v | --verbose" option
                      makeWindow{}: improved widget and menu code
                      added saveSize{} write_rc_file{} logFont{} changeFont{}
                      getField{}: changed regex to include "*" and "-"
    - client/ncid.conf-in: removed MsgFlag variable and comments
                           removed Raw variable and comments
                           added comments to CallOnRing section
                           added comments to Ring Options section
                           new "WakeUp" section
                           new "WrapLines" section
    - debian/Makefile: updated
    - debian/wc2ncid.init: new
    - debian/ncidsip.init: removed
    - doc/CONTRIBUTORS: updated
    - doc/NCID-SDK.odt: updated
    - doc/Setup-ncidsip: removed
    - doc/Setup-sip2ncid: updated
    - doc/Setup-wc2ncid: new
    - doc/INSTALL-Redhat: new
    - doc/INSTALL*: updated
    - doc/INSTALL-RPi: new
    - Fedora/Makefile: updated

modules/ncid-page.sh
#!/bin/sh

# ncid-page
# usage: ncid --no-gui --program ncid-page

# Last modified: Wed May 29, 2013

# sends Caller ID or message to a cell phone, pager, or any other email address
# Requires a mail program

# input is always 6 lines
#
# if input is from a call:
# input: DATE\nTIME\nNUMBER\nNAME\nLINE\nTYPE\n
#
# if input is from a message
# the message is in place of NAME:
# input: \n\n\n<MESSAGE>\n\nMSG\n

ConfigDir=/usr/local/etc/ncid/conf.d
ConfigFile=$ConfigDir/ncid-page.conf

# Test whether echo needs option "-e" to interpret new line "\n"
if [ "`echo -e`" != "-e" ] ; then use_e=" -e" ; else use_e= ; fi

# default mail program
PageMail=mail

# default email IP address
PageTo=

# default mail user if root is running script
PageFrom=mail

# default page subject option, either "" or "-s"
PageOpt=

# default $CIDTYPES to send page
PageTypes="CID"

[ -f $ConfigFile ] && . $ConfigFile

[ -z "PageTo" ] && \
{
    echo "Must set PageTo to SMS Gateway or other email address"
    exit 1
}

if [ -n "$PageFrom" ]
then
    # if valid user and script ID is root, set rootID
    [ "`id -nu $PageFrom 2> /dev/null`" = "$PageFrom" ] &&
    [ "`id -nu`" = "root" ] && rootID=1
fi

read CIDDATE
read CIDTIME
read CIDNMBR
read CIDNAME
read CIDLINE
read CIDTYPE

# Look for $CIDTYPE
for i in $PageTypes
do
    [ $i = "$CIDTYPE" ] && { found=1; break; }
done

# Exit if $CIDTYPE not found
[ -z "$found" ] && exit 0

if [ "$CIDTYPE" = "MSG" ]
then
    [ -n "$PageOpt" ] && PageOpt="$PageOpt \"Message\""
    MailMsg="\n$CIDNAME"
else
    [ -n "$PageOpt" ] && PageOpt="$PageOpt \"$CIDNMBR\""
    MailMsg="\nNCID TYPE: $CIDTYPE\nNAME: $CIDNAME\nNMBR: $CIDNMBR\nTIME: $CIDTIME\nDATE: $CIDDATE\n"
fi

# if line indicator found, include it
[ -n "$CIDLINE" ] && MailMsg="${MailMsg}LINE: $CIDLINE\n"

if [ -n "$rootID" ]
then
    # send mail as user $PageFrom
    echo $use_e $MailMsg |
        su -s /bin/sh -c "$PageMail $PageOpt $PageTo" $PageFrom
else
    # send mail as user running script
    echo $use_e $MailMsg | $PageMail $PageOpt $PageTo
fi

exit 0
